lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -158,21 +155,31 @@ class QcFlagService {
      * @return {Promise<void>} resolve once all periods are updated
      */
     async _updateEffectivePeriods({ from: newerPeriodFrom, to: newerPeriodTo }, periods) {
+        /**
+         * When run's timestamps are missing, effective periods have null timestamps, therefore
+         * comparison must be evaluated on extended set of values
+         * so  `from` == null becomes -1, which is interpreted as run.startTime
+         * and `to`   == null becomes MAX_DATE, which is interpreted as run.endTime
+         */
+        const MAX_DATE = new Date('100000-01-01').getTime();
+
```

Comment:
It's not because you put a comment explaining why that it becomes a good idea.
This makes the code VERY hard to understand. Please use explicit ifs, and use temporary variables to make things clear. In this case, directly using null will work, because `null <= 123` is true.

Then, for the effectiveTo, you can probably use something like
```js
const effectiveToIsBeforeNewerPeriodTo = effectiveTo === null
    || (newerPeriodTo !== null && effectievTo <= newerPeriodTo)
```

Reply:
applied